### ISSUE TITLE
allow disabling JS in development mode as well as production

### ISSUE THREAD

**Issue by usu** [nuxt/nuxt#22270]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v20.3.1
- Nuxt Version:     3.6.5
- Nitro Version:    2.5.2
- Package Manager:  npm@9.6.7
- Builder:          vite
- User Config:      devtools, devServer, experimental
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

Happy to deliver a reproduction if necessary for the discussion.

### Describe the bug

With experimental.noScripts = true we receive the following error in dev mode: "Cannot stringify arbitrary non-POJOs"

The message is correct in so far that we use non-POJOs as return values from useAsyncData.
However, it is surprising that data is hydrated at all in noScripts mode, as there is anyway no JS on client side to revive the data.

In production mode (nuxt preview), the error does not appear.

It seems noScripts setting is explicitly disabled in dev mode: https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/core/nitro.ts#L192
https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/core/runtime/nitro/renderer.ts#L331

What is the rationale for this? Shouldn't hydration be disabled completely, if noScripts is set (independent of dev or production mode)?

### Additional context

In Nuxt 2, the same concept worked properly with injectScripts = false (worked both in dev and production mode without the hydration error).

**Comment by danielroe** [comment]
Hm. This is a tricky one. I'd be happy with a setting to disable hydration in dev mode as well, as long as it wasn't on by default (because HMR is quite a nice feature when developing even if you intend for no scripts to be used in production).

**Comment by usu** [comment]
Yeah, I see. Wasn't aware hydration is a prerequisite for HMR, but kind of makes sense.

So you'd propose something like `options.experimental.noHydration`?

I also tried to override `options.dev`, but that had no effect at all.

**Comment by danielroe** [comment]
If your sole concern is the POJO issue, you could set ssr: false in development.

I would suggest having 'production', 'all', true , false as options for noScripts, and making current true option an alias for 'production'. (While deprecating true and advising moving to 'production'.)
